### DISCUSSION TITLE
External links to annotations

### DISCUSSION THREAD

**Discussion by rishkum** [pymupdf/PyMuPDF#1349] [Looking for help] [answered]
Hi 

So I have a dataframe with text that relates to the highlighted text in a PDF document. I was wondering if there is a programmatic way to have another column that takes you directly to the text by clicking a link?

**Comment by JorjMcKie** [comment, marked as answer]
What do you mean by dataframe? Pandas?
Whatever:
A textmarker annot does not by itself know anything about what it "marks". Its rectangle may happen to overlay text, but it could also be an image, drawing ... or nothing.
So you first have to determine what (text) the annot rect is wrapping. [Let's forget for the moment, that that text may be deleted or changed (via redactions) wihout the textmarker annot realizing this.]
Then you need some unique identifier for the annot. This must / should be a combination of its `/NM` key and the page ... number? Next problem: the page number may also change, if someone adds / deletes / rearranges pages.

If all this is not your concern, you can add that annot identifier to some other info store. There are annot lookup methods (per page) enough to then locate / load the annot.
